h2o-core/src/main/java/water/persist/PersistManager.java

```diff
@@ -268,7 +284,8 @@ else if(s.startsWith("s3:")) {
     else if( s.startsWith("hdfs:") ||
         s.startsWith("s3n:") || 
         s.startsWith("s3a:") || 
-        s.startsWith("maprfs:")) {
+        s.startsWith("maprfs:") ||
+        (useHdfsAsFallback() && I[Value.HDFS] != null && I[Value.HDFS].canHandle(s))) {
```

Comment:
This changes the current behavior. file:// will be handled by Hadoop instead of current I[Value.NFS].importFiles().